lib/services/platform-service.ts

```diff
@@ -217,6 +217,16 @@ export class PlatformService implements IPlatformService {
 			this.ensurePlatformInstalled(platform).wait();
 			let changesInfo = this.$projectChangesService.checkForChanges(platform);
 			if (changesInfo.hasChanges) {
+				// android build artifacts need to be cleaned up when switching from release to debug builds
+				if (platform.toLowerCase() === "android") {
```

Comment:
As the whole logic is Android specific, instead of using this if, you can move the whole code inside AndroidProjectService's cleanProject method. Here you can just call:
```TypeScript
platformData.platformProjectService.cleanProject(platformData.projectRoot, []).wait();
```